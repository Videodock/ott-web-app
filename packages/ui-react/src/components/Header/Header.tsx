import React, { type PropsWithChildren } from 'react';
import classNames from 'classnames';

import styles from './Header.module.scss';

type TypeHeader = 'static' | 'fixed';

type Props = {
  headerType?: TypeHeader;
  searchActive: boolean;
};

const Header = ({ children, headerType = 'static', searchActive }: PropsWithChildren<Props>) => {
  const headerClassName = classNames(styles.header, styles[headerType], {
    [styles.searchActive]: searchActive,
  });

  return (
    <header className={headerClassName}>
      <div className={styles.container}>{children}</div>
    </header>
  );
};
export default Header;
