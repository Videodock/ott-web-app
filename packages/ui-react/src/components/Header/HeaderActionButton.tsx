import React from 'react';
import classNames from 'classnames';

import IconButton, { type Props as IconButtonProps } from '../IconButton/IconButton';

import styles from './Header.module.scss';

const HeaderActionButton = ({ children, className, ...rest }: IconButtonProps) => {
  return (
    <IconButton className={classNames(styles.iconButton, className)} {...rest}>
      {children}
    </IconButton>
  );
};

export default HeaderActionButton;
