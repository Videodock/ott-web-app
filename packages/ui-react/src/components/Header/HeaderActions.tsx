import React, { type PropsWithChildren } from 'react';

import styles from './Header.module.scss';

const HeaderActions = ({ children }: PropsWithChildren) => {
  return <div className={styles.actions}>{children}</div>;
};

export default HeaderActions;
