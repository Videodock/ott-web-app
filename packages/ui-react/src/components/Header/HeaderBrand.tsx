import React from 'react';
import { useTranslation } from 'react-i18next';

import Logo from '../Logo/Logo';

import styles from './Header.module.scss';

type Props = {
  siteName?: string;
  logoSrc?: string | null;
  setLogoLoaded: (loaded: boolean) => void;
};

const HeaderBrand = ({ siteName, logoSrc, setLogoLoaded }: Props) => {
  const { t } = useTranslation('menu');

  if (!logoSrc) return null;

  return (
    <div className={styles.brand}>
      <Logo alt={t('logo_alt', { siteName })} src={logoSrc} onLoad={() => setLogoLoaded(true)} />
    </div>
  );
};

export default HeaderBrand;
