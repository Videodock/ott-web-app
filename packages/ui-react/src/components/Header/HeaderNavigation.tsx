import React from 'react';

import Button from '../Button/Button';

import styles from './Header.module.scss';

type NavItem = {
  label: string;
  to: string;
};

const HeaderNavigation = ({ navItems }: { navItems: NavItem[] }) => {
  return (
    <nav className={styles.nav}>
      <ul>
        {navItems.map((item, index) => (
          <li key={index}>
            <Button activeClassname={styles.navButton} label={item.label} to={item.to} variant="text" />
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default HeaderNavigation;
