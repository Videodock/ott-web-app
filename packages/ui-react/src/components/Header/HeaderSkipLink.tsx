import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './Header.module.scss';

const HeaderSkipLink = () => {
  const { t } = useTranslation('menu');
  return (
    <a href="#content" className={styles.skipToContent}>
      {t('skip_to_content')}
    </a>
  );
};

export default HeaderSkipLink;
