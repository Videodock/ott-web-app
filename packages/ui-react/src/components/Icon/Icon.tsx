import classNames from 'classnames';
import React, { FC, SVGProps } from 'react';

import styles from './Icon.module.scss';

export type Props = {
  icon: FC<SVGProps<SVGSVGElement>>;
  className?: string;
};

const Icon: FC<Props> = ({ icon: IconComponent, className }) => {
  return <IconComponent className={classNames(styles.icon, className)} aria-hidden="true" />;
};

export default Icon;
