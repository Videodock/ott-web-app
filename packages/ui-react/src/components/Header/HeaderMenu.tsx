import React from 'react';
import { useTranslation } from 'react-i18next';
import Menu from '@jwp/ott-theme/assets/icons/menu.svg?react';

import Icon from '../Icon/Icon';
import IconButton from '../IconButton/IconButton';

import styles from './Header.module.scss';

const HeaderMenu = ({ sideBarOpen, onClick }: { sideBarOpen: boolean; onClick: () => void }) => {
  const { t } = useTranslation('menu');

  return (
    <div className={styles.menu}>
      <IconButton className={styles.iconButton} aria-label={t('open_menu')} aria-expanded={sideBarOpen} onClick={onClick}>
        <Icon icon={Menu} />
      </IconButton>
    </div>
  );
};

export default HeaderMenu;
