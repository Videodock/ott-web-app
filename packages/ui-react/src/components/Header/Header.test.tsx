import React from 'react';
import { render } from '@testing-library/react';
import Language from '@jwp/ott-theme/assets/icons/language.svg?react';

import Icon from '../Icon/Icon';

import Header from './Header';
import HeaderSkipLink from './HeaderSkipLink';
import HeaderMenu from './HeaderMenu';
import HeaderBrand from './HeaderBrand';
import HeaderNavigation from './HeaderNavigation';
import HeaderActions from './HeaderActions';
import HeaderActionButton from './HeaderActionButton';

vi.mock('react-router-dom', () => ({
  NavLink: () => 'a',
  Link: () => 'a',
}));

describe('<Header />', () => {
  test('renders header', () => {
    const { container } = render(
      <Header searchActive={false}>
        <HeaderSkipLink />
        <HeaderMenu onClick={vi.fn()} sideBarOpen={false} />
        <HeaderBrand logoSrc={undefined} setLogoLoaded={vi.fn()} siteName="OTT Web App" />
        <HeaderNavigation navItems={[]} />
        <HeaderActions>
          <HeaderActionButton>
            <Icon icon={Language} />
          </HeaderActionButton>
        </HeaderActions>
      </Header>,
    );

    expect(container).toMatchSnapshot();
  });

  test('renders header with nav buttons', () => {
    const navItems = [
      { label: 'Home', to: '/' },
      { label: 'Button test', to: '/test' },
    ];
    const { container } = render(
      <Header searchActive={false}>
        <HeaderSkipLink />
        <HeaderMenu onClick={vi.fn()} sideBarOpen={false} />
        <HeaderBrand logoSrc="/logo.png" setLogoLoaded={vi.fn()} siteName="OTT Web App" />
        <HeaderNavigation navItems={navItems} />
        <HeaderActions>
          <HeaderActionButton>
            <Icon icon={Language} />
          </HeaderActionButton>
        </HeaderActions>
      </Header>,
    );

    expect(container).toMatchSnapshot();
  });
});
