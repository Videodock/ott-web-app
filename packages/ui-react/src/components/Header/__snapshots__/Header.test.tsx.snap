// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > renders header 1`] = `
<div>
  <header
    class="_header_f4f7a7 _static_f4f7a7"
  >
    <div
      class="_container_f4f7a7"
    >
      <a
        class="_skipToContent_f4f7a7"
        href="#content"
      >
        skip_to_content
      </a>
      <div
        class="_menu_f4f7a7"
      >
        <button
          aria-expanded="false"
          aria-label="open_menu"
          class="_iconButton_0fef65 _iconButton_f4f7a7"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="_icon_585b29"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </button>
      </div>
      <nav
        class="_nav_f4f7a7"
      >
        <ul />
      </nav>
      <div
        class="_actions_f4f7a7"
      >
        <button
          class="_iconButton_0fef65 _iconButton_f4f7a7"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="_icon_585b29"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </header>
</div>
`;

exports[`<Header /> > renders header with nav buttons 1`] = `
<div>
  <header
    class="_header_f4f7a7 _static_f4f7a7"
  >
    <div
      class="_container_f4f7a7"
    >
      <a
        class="_skipToContent_f4f7a7"
        href="#content"
      >
        skip_to_content
      </a>
      <div
        class="_menu_f4f7a7"
      >
        <button
          aria-expanded="false"
          aria-label="open_menu"
          class="_iconButton_0fef65 _iconButton_f4f7a7"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="_icon_585b29"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </button>
      </div>
      <div
        class="_brand_f4f7a7"
      >
        a
      </div>
      <nav
        class="_nav_f4f7a7"
      >
        <ul>
          <li>
            a
          </li>
          <li>
            a
          </li>
        </ul>
      </nav>
      <div
        class="_actions_f4f7a7"
      >
        <button
          class="_iconButton_0fef65 _iconButton_f4f7a7"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="_icon_585b29"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </header>
</div>
`;
